ansible_kernel/kernel.py

```diff
@@ -195,8 +198,84 @@ def clean_up_task_files(self):
                 os.unlink(task_file)
         self.task_files = []
 
+    def runner_process_message(self, data):
+        logger.info("runner message:\n{}".format(pprint.pformat(data)))
+        try:
+
+            event_data = data.get('event_data', {})
+            task = event_data.get('task')
+            event = data.get('event')
+
+            if event == 'playbook_on_start':
```

Comment:
I think I see what you're doing with these events now... it needs to be pretty selecting on what it shows?

Reply:
UI customization is done through the kernel.   I want to show more than ansible-playbook does on the command line, but less than is in the event structure.   The first use-case for this is for self-directed learning.  So I want to expose some of the data structures but not too much.   

In the future I plan to make use of Jupyter Widgets for event better customization.